machines/biometrics.ts

```diff
@@ -0,0 +1,227 @@
+      init: {
+        invoke: {
+          src: LocalAuthentication.hasHardwareAsync,
+          onError: 'failure',
+          onDone: {
+            target: 'initAuthTypes',
+            actions: assign({ isAvailable: (context, event) => event.data }),
+          }
+        }
+      },
+
+      initAuthTypes: {
+        invoke: {
+          src: LocalAuthentication.supportedAuthenticationTypesAsync,
+          onError: 'failure',
+          onDone: {
+            target: 'initEnrolled',
+            actions: assign({ authTypes: (context, event) => event.data }),
+          }
+        }
+      },
+
+      initEnrolled: {
+        invoke: {
+          src: LocalAuthentication.isEnrolledAsync,
+          onError: 'failure',
+          onDone: {
+            target: 'checking',
+            actions: assign({ isEnrolled: (context, event) => event.data }),
+          }
+        }
+      },
+
+      // Checks whether we need to proceed if its available otherwise it gets to failure
+      checking: {
+        always: [
+          {
+            target: 'available',
+            cond: 'checkIfAvailable'
+          },
+          {
+            target: 'failure.unavailable',
+            cond: 'checkIfUnavailable'
+          },
+          {
+            target: 'failure.unenrolled',
+            cond: 'checkIfUnenrolled'
+          }
+        ]
+      },
+
+
+      // if available then wait for any event
+      available: {
+        on: {
+          AUTHENTICATE: 'authenticating'
+        }
+      },
+
+      // authenticating biometrics
+      authenticating: {
+        invoke: {
+          src: () => async () => {
+            let res = await LocalAuthentication.authenticateAsync({
+              promptMessage: 'Biometric Authentication',
+
+              // below can only works for IOS not android
+              // disableDeviceFallback: true,
+              // fallbackLabel: 'Invalid fingerprint attempts, Please try again.'
+            })
+            return res.success;
+          },
+          onError: 'failure',
+          onDone: {
+            target: 'authentication',
+            actions: assign({ status: (context, event) => event.data })
+          }
+        },
+      },
+
+      reauthenticating: {
+        always: [
+          {
+            target: 'authenticating',
+            cond: 'checkIfAvailable'
+          },
+          {
+            target: 'failure.unenrolled'
+          }
+        ]
+      },
+
+      // checks authentication status
+      authentication: {
+        always: [
+          {
+            target: 'success',
+            cond: ctx => ctx.status
+          },
+          {
+            target: 'failure.failed',
+            cond: ctx => !ctx.status
+          }
+        ]
+      },
+
+      success: {
+        meta: 'Success authentication with Biometrics',
+      },
+
+      failure: {
+        initial: 'error',
+        states: {
+          unavailable: {
+            meta: 'Device does not support Biometrics'
```

Comment:
Use an object for `meta` and use descriptive property names like "message", "description", etc.

Reply:
aha! originally I used "message" object but had warning issue with vscode. hehhe, then I found out you are using object meta, so for consistency ill change this up :) thanks